src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,299 @@
+    const std::string& module_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const std::map<std::string, std::string> params = {
+      {enums_value::kRadio, strings::kradioControlCapabilities},
+      {enums_value::kClimate, strings::kclimateControlCapabilities},
+      {enums_value::kAudio, strings::kaudioControlCapabilities},
+      {enums_value::kLight, strings::klightControlCapabilities},
+      {enums_value::kHmiSettings, strings::khmiSettingsControlCapabilities},
+      {enums_value::kSeat, strings::kseatControlCapabilities}};
+  bool is_module_type_valid = false;
+  for (const auto& param : params) {
+    if (param.first == module_type) {
+      if (rc_capabilities.keyExists(param.second)) {
+        is_module_type_valid = true;
+      }
+    }
+  }
+  return is_module_type_valid;
+}
+
+void GetInteriorVehicleDataRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject* rc_capabilities =
+      hmi_capabilities_.rc_capability();
+
+  const std::string module_type = ModuleType();
+  if (rc_capabilities &&
+      !CheckIfModuleTypeExistInCapabilities(*rc_capabilities, module_type)) {
+    LOG4CXX_WARN(logger_, "Accessing not supported module data");
+    SetResourceState(ModuleType(), ResourceState::FREE);
+    SendResponse(false,
+                 mobile_apis::Result::UNSUPPORTED_RESOURCE,
+                 "Accessing not supported module data");
+    return;
+  }
+
+  app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(connection_key());
+
+  if (HasRequestExcessiveSubscription()) {
+    excessive_subscription_occured_ = true;
+    is_subscribed =
+        (*message_)[app_mngr::strings::msg_params][message_params::kSubscribe]
+            .asBool();
+    RemoveExcessiveSubscription();
+  }
+
+  (*message_)[app_mngr::strings::msg_params][app_mngr::strings::app_id] =
+      app->app_id();
+
+  SendHMIRequest(hmi_apis::FunctionID::RC_GetInteriorVehicleData,
+                 &(*message_)[app_mngr::strings::msg_params],
+                 true);
+}
+
+void GetInteriorVehicleDataRequest::on_event(
+    const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  RCCommandRequest::on_event(event);
+
+  if (hmi_apis::FunctionID::RC_GetInteriorVehicleData != event.id()) {
+    return;
+  }
+
+  smart_objects::SmartObject& hmi_response =
+      const_cast<smart_objects::SmartObject&>(event.smart_object());
+
+  mobile_apis::Result::eType result_code =
+      GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
+          hmi_response[app_mngr::strings::params][app_mngr::hmi_response::code]
+              .asUInt()));
+
+  bool result =
+      helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result_code,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS);
+
+  if (mobile_apis::Result::READ_ONLY == result_code) {
+    result = false;
+    result_code = mobile_apis::Result::GENERIC_ERROR;
```

Comment:
why translate mobile_apis::Result::READ_ONLY to mobile_apis::Result::GENERIC_ERROR?